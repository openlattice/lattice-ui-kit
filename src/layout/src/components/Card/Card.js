// @flow

import styled, { css } from 'styled-components';
import * as Colors from '../../../../colors';

const { NEUTRALS, WHITE } = Colors;

type CardProps = {
  onClick :Function;
}

const getHoverStyles = ({ onClick } :CardProps) => {
  if (onClick) {
    return css`
      :hover {
        box-shadow: 0 2px 8px -2px rgba(0, 0, 0, 0.15);
        cursor: pointer;

        * {
          cursor: inherit;
          pointer-events: auto;
        }
      }

      :active {
        box-shadow: none;
      }
    `;
  }

  return null;
};

const Card = styled.div`
  background-color: ${WHITE};
  border: 1px solid ${NEUTRALS[4]};
  border-radius: 5px;
  display: flex;
  flex: 0 0 auto;
  flex-direction: column;
  position: relative;

  & > div {
    border-bottom: 1px solid ${NEUTRALS[4]};
  }

  & > div:first-child {
    border-radius: 5px 5px 0 0;
  }

  & > div:last-child {
    border-bottom: 0;
    border-radius: 0 0 5px 5px;
  }

  ${getHoverStyles};
`;

export default Card;
