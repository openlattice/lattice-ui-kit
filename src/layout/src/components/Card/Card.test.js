import toJson from 'enzyme-to-json';
import { mount, shallow } from 'enzyme';

import Card from './Card';

describe('Card', () => {

  describe('snapshots', () => {

    test('Without Children', () => {
      const wrapper = mount(<Card />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('With child divs', () => {
      const wrapper = mount(
        <Card>
          <div>1</div>
          <div>2</div>
          <div>3</div>
        </Card>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('render', () => {

    test('should render div', () => {
      const wrapper = mount(<Card />);
      expect(wrapper.find('div')).toHaveLength(1);
    });

    test('should include all children in the render tree', () => {
      const children = (
        <div id="test-id">
          <h2>
            heading
          </h2>
          <ul className="list">
            <li>
              one
            </li>
          </ul>
          <p>
            this is a paragraph
          </p>
          <button type="button">
            clicky 1
          </button>
        </div>
      );
      const wrapper = shallow(
        <Card>
          { children }
        </Card>
      );
      expect(wrapper.contains(children)).toEqual(true);
    });

  });

});
