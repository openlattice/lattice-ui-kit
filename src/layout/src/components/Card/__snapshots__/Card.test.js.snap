// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshots With child divs 1`] = `
<ForwardRef>
  <WithStyles(ForwardRef(Card))
    variant="outlined"
  >
    <ForwardRef(Card)
      classes={
        Object {
          "root": "MuiCard-root",
        }
      }
      variant="outlined"
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiCard-root"
        elevation={1}
        variant="outlined"
      >
        <ForwardRef(Paper)
          className="MuiCard-root"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={1}
          variant="outlined"
        >
          <div
            className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
          >
            <div>
              1
            </div>
            <div>
              2
            </div>
            <div>
              3
            </div>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</ForwardRef>
`;

exports[`Card snapshots Without Children 1`] = `
<ForwardRef>
  <WithStyles(ForwardRef(Card))
    variant="outlined"
  >
    <ForwardRef(Card)
      classes={
        Object {
          "root": "MuiCard-root",
        }
      }
      variant="outlined"
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiCard-root"
        elevation={1}
        variant="outlined"
      >
        <ForwardRef(Paper)
          className="MuiCard-root"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={1}
          variant="outlined"
        >
          <div
            className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
          />
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</ForwardRef>
`;
